File: pyutil/inface.cpp
#include "inface.h"
#include "parsepy.h"
#include "../core/util.h"
#include <exception> 
#include <string.h> 
using namespace std;

int test(char *rstr, int size)
{
	try
	{
		parsepy py;

		//import py moudle
		py.runpy((char *)"import sys");
		py.runpy((char *)"sys.path.append('..')");

		//open file
		pyobject *obj[1] = {py.pysfs((char *)"server.ini")};
		pyobject *argtuple = py.setargtuple(obj, 1);
		pyobject *ret = py.exem((char *)"pyutil.pconfig", (char *)"openconfig", argtuple);
		pyobject *fp = py.parseobj(ret);

		//get key value
		pyobject *obj4[4] = {fp, py.pysfs((char *)"NET"),
		 py.pysfs((char *)"ip"), py.pysfs((char *)"127.0.0.1")};
		argtuple = py.setargtuple(obj4, 4);
		ret = py.exem((char *)"pyutil.pconfig", (char *)"pastring", argtuple);
		strncpy(rstr, py.parsestr(ret), size);

		//close file
		argtuple = py.newtuple(1);
		py.intuple(argtuple, 0, fp);
		ret = py.exem((char *)"pyutil.pconfig", (char *)"closeconfig", argtuple);
	}
	catch (exception &e)
	{
		strncpy(rstr, e.what(), size);
		return FAILED;
	}

	return SUCCESS;
}

int getsyscon(const char *filename, sys *sysc)
{	
	try
	{
		parsepy py;

		//import py moudle
		py.runpy((char *)"import sys");
		py.runpy((char *)"sys.path.append('..')");

		pyobject *sysclass = py.insclass((char *)"pyutil.inface", (char *)"sys", NULL, NULL); 

		//read config
		pyobject *obj[2] = {py.pysfs((char *)filename), sysclass};
		pyobject *argtuple = py.setargtuple(obj, 2);
		pyobject *ret = py.exem((char *)"pyutil.inface", (char *)"getsyscon", argtuple);

		//获取类的成员
		pyobject *ip = py.callmethod(sysclass, (char *)"getip");
		char *cip = py.parsestr(ip);
		strncpy((char *)sysc->ip, cip, sizeof(sysc->ip));

		pyobject *port = py.callmethod(sysclass, (char *)"getport");
		sysc->port = py.parseint(port);
	}
	catch (exception &e)
	{
		return FAILED;
	}

	return SUCCESS;	
}

File: pyutil/inface.py
#!/usr/bin/python
#class.py

import pconfig as pcon

class sys :
	def __init__(self) :
		self.ip = ""
		self.port = 0

	def getip(self) :
		return self.ip

	def getport(self) :
		return self.port

	def setip(self, ip) :
		self.ip = ip

	def setport(self, port) :
		self.port = port

def getsyscon(filename, sysc) :
	f = pcon.openconfig(filename)
	if f == None :
		return 0

	ip = pcon.pastring(f, "NET", "ip", "127.0.0.1")
	sysc.setip(ip)

	port = pcon.paint(f, "NET", "port", 8888)
	sysc.setport(port)

	pcon.closeconfig(f)

	return 1

if __name__ == '__main__' :
	sysc = sys()
	ret = getsyscon('../conf/server.ini', sysc)
	if (ret == 1) :
		print sysc.getip(), sysc.getport()

